### PR TITLE
Adjust synthetic dataset generator to use Merlin dtypes (in schemas etc)

### PR DESCRIPTION
**Pull request by karlhigley**
Depends on https://github.com/NVIDIA-Merlin/core/pull/170

### Goals :soccer:
Adjusts this library to use a new set of standard dtypes for Merlin in schemas.

### Implementation Details :construction:
We expect this to help us smooth out some of the cross-library communication and dtype translation issues we've encountered. It also provides a way to clearly delineate which dtypes we do and we do not support Merlin-wide.

### Testing Details :mag:
Covered by existing tests since this is purely a refactor.

### CODE REVIEW

**Comment by karlhigley** [comment]
This relies on breaking changes in https://github.com/NVIDIA-Merlin/core/pull/170, but I don't see a clear way to run the tests against a particular version/branch of Core. I updated the `tox.ini`, but that doesn't seem to be what's driving most of the test runs.

**Comment by sararb** [comment]
Changes looks fine, we will merge it after https://github.com/NVIDIA-Merlin/core/pull/170 is merged

File: merlin/datasets/entertainment/music_streaming/schema.json [hunk 1]
```diff
@@ -160,6 +160,7 @@
     },
     {
       "name": "click",
+      "type": "INT",
```
[the remark this answers is not in the document]
Comment (karlhigley) [reply merlin/datasets/entertainment/music_streaming/schema.json:163]: Yeah, exactly! Centralizing the dtype code gives us a way to raise a consistent error all over the Merlin code base when client code tries to do funky things with dtypes, which is how I found these missing ones.

**Comment by karlhigley** [comment]
@sararb The NVT and Systems failures are expected here, since the tests for those libraries depend both on the changes in this PR and on updztes to use the new dtypes from Core that we won't be able to merge until after this PR goes through. If you're good with it, let's go ahead and merge this so we can get the other libraries updated too.

**Review by sararb — APPROVED** [review]

**Pull request merged**